converter/shieldhit/geo.py

```diff
@@ -0,0 +1,189 @@
+from converter.solid_figures import SolidFigure, BoxFigure, CylinderFigure, SphereFigure
+from abc import ABC
+from dataclasses import dataclass, field
+from math import log10, ceil
+
+
+def format_float(number: float, n: int) -> float:
+    """
+    Format float to be up to n characters wide, as precise as possible and as short
+    as possible (in descending priority). so for example given 12.333 for n=5 you will
+    get 12.33, n=7 will be 12.333
+    """
+    # If number is zero we just want to get 0.0 (it would mess up the log10 operation below)
+    if number == 0:
+        return 0.0
+
+    # Adjust n for .
+    n -= 1
```

Comment:
It is a bad idea to modify parameters, for the reader it is not clear whether the parameter will be modified upon function exit or not, so better to avoid it.
Make a new variable with longer name, assign it to n, then modify accordingly.